core/libp2p/multi/multihash.hpp

```diff
@@ -6,13 +6,80 @@
 #ifndef KAGOME_MULTIHASH_HPP
 #define KAGOME_MULTIHASH_HPP
 
+#include <cstdint>
+#include <stdexcept>
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
 namespace libp2p::multi {
+
   /**
    * Special format of hash used in Libp2p. Allows to differentiate between
    * outputs of different hash functions. More
    * https://github.com/multiformats/multihash
    */
-  class Multihash {};
+  class Multihash {
+    using Hash = kagome::common::Buffer;
+    using HashType = std::uint8_t;  // may be later changed to a structure
+                                    // containing info about type
+
+   public:
+    /**
+     * @brief Creates a multihash from hash type and the hash itself. Note that
+     * the max hash length is 127
+     * @param type - numerical code of the hash type.
+     * @param hash - binary buffer with the hash
+     * @return Value with the multihash in case of success, Error otherwise
+     */
+    static auto create(const HashType &type, const Hash &hash)
+        -> kagome::expected::Result<Multihash, std::invalid_argument>;
+
+    /**
+     * @brief Creates a multihash from a string, which represents a binary
+     * buffer in hexadecimal form. The first byte denotes the hash type, the
+     * second one contains the hash length, and the following are the hash
+     * itself
+     * @param hex - the string with hexadecimal representation of the multihash
+     * @return Value with the multihash in case of success, Error otherwise
+     */
+    static auto createFromHex(std::string_view hex)
+        -> kagome::expected::Result<Multihash, std::invalid_argument>;
+
+    /**
+     * @return the info about hash type
+     */
+    const HashType &getType() const;
+
+    /**
+     * @return the hash stored in this multihash
+     */
+    const Hash &getHash() const;
```

Comment:
@kamilsa do we use snake_case or camelCase?

Reply:
For functions we use camelCase